tf_transform/tf_transform/tf_transform_node.py

```diff
@@ -0,0 +1,61 @@
+import rclpy
+from rclpy.node import Node
+import tf2_ros
+import geometry_msgs.msg as gm
+
+class TFTransformNode(Node):
+
+    def __init__(self):
+        super().__init__('tf_transform')
+        self.tf_buffer = tf2_ros.Buffer()
+        self.tf_listener = tf2_ros.TransformListener(self.tf_buffer, self)
+        self.tf_broadcaster = tf2_ros.TransformBroadcaster(self)
+
+    def transform_callback(self):
+        try:
+                # Look up the transformations between "map" and "base_link"
+            trans_map = self.tf_buffer.lookup_transform("map", "base_link", rclpy.time.Time())
+            trans_base = self.tf_buffer.lookup_transform("base_link", "base_link", rclpy.time.Time())  # Identity transformation
+            
+            translation_map = trans_map.transform.translation
+            rotation_map = trans_map.transform.rotation
+            
+            translation_base = trans_base.transform.translation
+            
+            # Set the desired translation by keeping the x and y components from "map"
```

Comment:
Explain what you are doing in the comments. `trans_map` is the transformation that takes points in the base frame and outputs points in the map frame. The translation component of the transform is thus the position of the robot relative to the map origin, defined in the map frame. Setting the z component to 0 creates a pseudo robot frame that is a projection of the true position onto the map xy plane.